### PR TITLE
improve accumulator append

### PR DESCRIPTION
**Pull request by nkysg**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?
improve accumulator append

Issue Number: N/A

## What is the new behavior?

-
-
-

### CODE REVIEW

**Comment by nkysg** [comment]
@0xpause

File: commons/accumulator/src/tree.rs [hunk 1]
```diff
@@ -265,6 +296,9 @@ impl AccumulatorTree {
         // get hash from cache
         let mut temp_index = index;
         let mut index_key = temp_index;
+        if let Some(node_hash) = self.index_frozen_subtrees.get(&index_key) {
```
[the remark this answers is not in the document]
Comment (nkysg) [reply commons/accumulator/src/tree.rs:299]: 有用，这个只是append算节点的时候不需要遍历查找，get_leaves和get_proof都需要用get_node_hash_always

**Review by jolestar — APPROVED** [review]
LGTM

File: commons/accumulator/src/tree.rs [hunk 2]
```diff
@@ -29,6 +29,12 @@ pub struct AccumulatorTree {
     pub(crate) store: Arc<dyn AccumulatorTreeStore>,
     /// The temp update nodes
     update_nodes: HashMap<HashValue, AccumulatorNode>,
+    /// index of frozen_subtree_roots
+    /// max size will be 64
+    index_frozen_subtrees: HashMap<NodeIndex, HashValue>,
+    /// index_to_freeze NodeIndex => HashValue
+    /// left than max_to_freeze
```
Comment (0xpause) [inline commons/accumulator/src/tree.rs:36]: "left than" 这里是指  less than ?

Comment (nkysg) [reply commons/accumulator/src/tree.rs:36]: fixed

**Review by 0xpause — APPROVED** [review]

**Comment by nkysg** [comment]
https://github.com/starcoinorg/starcoin/commit/4261672c4e4efa62d8680d3a089fcaa7e3719c5a 做了#3551的对比测试，导入main和barnard的快照，目前没有问题

**Pull request merged**